### PR TITLE
Add cryptocurrency

### PR DESCRIPTION
**Pull request by elsesiy**
This PR adds the possibility to convert a cryptocurrency to a regular currency almost similar to the currency script (which I used for all the boilerplate). 

For now I added the top 10 cryptocurrencies as of today. I think I updated all necessary files to adhere to your repo structure.

If you have any remarks, feel free to comment.

Cheers, Jonas

### CODE REVIEW

**Comment by alexanderepstein** [comment]
I love the idea! 

A few things:
* It might make more sense to integrate this into currency rather than a standalone tool
* The user should be able to convert from standard currencies to cryptocurrencies (two way conversion)
* ```pcregrep``` needs to be replaced with ```grep -Eo``` as ```pcregrep``` is not standard on all systems (including mine)
* By using the existing ```currency``` converter you should be able to convert cryptocurrencies to more standard currencies. Process would be convert cryptocurrency to USD and then use currency to convert USD to whatever the true target currency is. This would also work for converting to cryptocurrencies by first converting the base currency to USD and then converting that to the target cryptocurrency.

**Comment by elsesiy** [comment]
@alexanderepstein Thanks you for your remarks. Of course I was thinking about integrating this functionality into currency but decided against it. 

- The primary reason to create an own script is that I'm planning on extending the module to also list i.e. the top10 currencies, see relative change in rank and stuff like this which is exclusive to and of particular interest only for cryptocurrencies. If I were to include this into currency, it's going to be a little ugly in terms of usability.

- I'm happy to replace pcregrep with grep -Eo but I haven't found a possibility to use only the second matching group with regular grep.

- I'll get rid of `story.bash`, `sparrow.json` and `sparrow.md`

**Comment by alexanderepstein** [comment]
That makes total sense, if that is the plan this can stay as a standalone tool.
You should be able to use ```grep -Eo``` and by setting the variable as an array and the line after set the variable to the second index of that array ex. If you have a text file that looks like this:
```bash
hello
hel
```
then you could use grep like this
```bash
variable=($(cat text.txt | grep -Eo hel[A-z]*)) # notice the extra parenthesis this says create an array
variable=${variable[1]} ## grabs the second result
echo $variable # this should print out hel
```

I also changed the readme so that all of the tools are in alphabetical order (this is one of the reason you see conflicts)
I also changed the installer so that this could be added to homebrew core. If you copy paste what you added to the readme here then merge with the master branch to get rid of the conflicts (don't use any of your changes to these two files) and i will add back what you changed on the readme after it has been merged and i will add cryptocurrencies to the install

File: cryptocurrency/cryptocurrency [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+checkInternet()
+{
+  httpGet google.com > /dev/null 2>&1 || { echo "Error: no active internet connection" >&2; return 1; } # query google with a get request
+}
+
+## Grabs the exchange rate and does the math for converting the currency
+convertCurrency()
+{
+  exchangeTo=$(echo "$exchangeTo" | tr '[:upper:]' '[:lower:]')
+  exchangeRate=$(httpGet "https://api.coinmarketcap.com/v1/ticker/$reqId/?convert=$exchangeTo" | pcregrep -o2 "(\"price_$exchangeTo\")\s*:\s*.([0-9]*[.][0-9]*)") > /dev/null
```
Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: Change this line to ```httpGet "https://api.coinmarketcap.com/v1/ticker/$reqId/?convert=$exchangeTo" | grep -Eo '"price_$exchangeTo": "[0-9 .]*' | grep -Eo  '[0-9 .]*'``` and that should replace the use of pcregrep

Comment (elsesiy) [inline cryptocurrency/cryptocurrency]: Unfortunately this doesn't work as it results in a stdin parse error. I also tried to use pure bash regex matching with `=~` and capture the group via `"${BASH_REMATCH[1]}"` but no luck so far...

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: I actually have a syntax error in there that probably caused this give me a second to rectify.

Comment (elsesiy) [inline cryptocurrency/cryptocurrency]: Yes I saw it and changed it to `exchangeRate=$(httpGet "https://api.coinmarketcap.com/v1/ticker/$reqId/?convert=$exchangeTo") > /dev/null | grep -Eo "\"price_$exchangeTo\": \"[0-9.]*" | grep -Eo "[0-9.]*"` but still no luck.

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: Why did you add the -N flag to ```curl``` btw?

Comment (elsesiy) [inline cryptocurrency/cryptocurrency]: Because afaik curl buffers the output and outputs the chunks by default. For this particular case this doesn't make any sense right? See [here](https://curl.haxx.se/docs/manpage.html#-N).

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: It doesn't really matter here since we redirect the output of curl into the variable so the user doesn't see it. I fixed it I forgot that you need double quotes around strings with variables. 
```bash
exchangeRate=$(httpGet "https://api.coinmarketcap.com/v1/ticker/$reqId/?convert=$exchangeTo" | grep -Eo "\"price_$exchangeTo\": \"[0-9 .]*" | grep -Eo "[0-9 .]*") > /dev/null
```

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: For some reason though when it goes to print the exchange rate it goes onto a newline for some reason.

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: Terrible work around but adding this line afterwards gets rid of the newline error ```exchangeRate=$( echo "$exchangeRate * 1" | bc)```

**Comment by alexanderepstein** [comment]
I never got your feedback on two way conversion, what do you think?

File: cryptocurrency/cryptocurrency (hunk 1 above)
Comment (elsesiy) [inline cryptocurrency/cryptocurrency]: This still doesn't on my end (OS X, zsh). Still the same error `(standard_in) 1: parse error`

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: Huh let me pop onto my mac and see if i can troubleshoot.

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]:
```bash
exchangeRate=$(httpGet "https://api.coinmarketcap.com/v1/ticker/$reqId/?convert=$exchangeTo" | grep -Eo "\"price_usd\": \"[0-9 .]*" | sed s/"\"price_usd\": \""//g) > /dev/null
exchangeRate=$( echo "$exchangeRate * 1" | bc)
```
Let me know if that works on yours, the old command didn't work on my mac either (probably cause they are still using bash 3) but this one does.

Comment (elsesiy) [inline cryptocurrency/cryptocurrency]: Looks good and there's no need to `bc` on my machine... Shall I include it anyway?

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: Let me check if its necessary now that I am using sed it seemed odd to me that it would need to be there.

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: No need for it definitely get rid of it 😄

Comment (elsesiy) [inline cryptocurrency/cryptocurrency]: Great so all remarks resolved for now right? 👍

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: Are you planning on adding [this](https://github.com/alexanderepstein/Bash-Snippets/pull/88#issuecomment-318473398) or no? I will still probably merge this for now if you don't want to do this immediately, you can always add this functionality later.

Comment (elsesiy) [inline cryptocurrency/cryptocurrency]: I'm looking into this probably this weekend but no promises :) I guess it's fine to add this now and enhance it later?

Comment (alexanderepstein) [inline cryptocurrency/cryptocurrency]: Indeed it is, going to merge now 🎉
